tests/unit/preprocessor/RandomFourierGaussPreproc_unittest.cc

```diff
@@ -0,0 +1,69 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Author: Nanubala Gnana Sai
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/kernel/GaussianKernel.h>
+#include <shogun/mathematics/NormalDistribution.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/preprocessor/DensePreprocessor.h>
+#include <shogun/preprocessor/RandomFourierGaussPreproc.h>
+
+using namespace shogun;
+
+class RFGPTest : public ::testing::Test
+{
+public:
+	virtual void SetUp()
+	{
+		SGMatrix<float64_t> mat(num_features, num_vectors);
+
+		linalg::range_fill(mat, 1.0);
+
+		auto features = std::make_shared<DenseFeatures<float64_t>>(mat);
+		preproc = std::make_shared<RandomFourierGaussPreproc>();
+		preproc->put("m_dim_output", target_dim);
+		preproc->fit(features);
+	}
+	virtual void TearDown()
+	{
+	}
+
+protected:
+	const int32_t seed = 100;
+	const index_t num_vectors = 5;
+	const index_t num_features = 3;
+	const index_t target_dim = 100;
+	std::shared_ptr<RandomFourierGaussPreproc> preproc;
+};
+
+TEST_F(RFGPTest, approx)
+{
+	SGVector<float64_t> vec_1(num_features);
+	SGVector<float64_t> vec_2(num_features);
+
+	std::mt19937_64 prng(seed);
+
+	NormalDistribution<float64_t> norm_dist;
+
+	for (auto idx : range(num_features))
+	{
+		vec_1[idx] = norm_dist(prng);
+		vec_2[idx] = norm_dist(prng);
+	}
+
+	auto processed_vec1 = preproc->apply_to_feature_vector(vec_1);
+	auto processed_vec2 = preproc->apply_to_feature_vector(vec_2);
+
+	auto result_rff = linalg::dot(processed_vec1,processed_vec2);
+	auto feat_1 = std::make_shared<DenseFeatures<float64_t>>(vec_1);
+	auto feat_2 = std::make_shared<DenseFeatures<float64_t>>(vec_2);
+	auto gauss_kernel = std::make_shared<GaussianKernel>();
```

Comment:
could you pls use a non-trivial bandwidth? (You will need to match it with the rff) You were lucky that the test works because the bandwidths seem to be the same just by chance. But it is better to make that explicit

Reply:
Actually, the methods for vector/matrix are specific to the class here. Could you add a test against the public API of the transformer baseclass? I.e. the apply method?

Reply:
I think you mean transform?

Reply:
The difference is really coming out to be erratic, here's the report for increasing num dims :
```
100 1.0047037237446161
200 0.98600405839838312
300 0.96979195105536831
500  0.98728862707465515
1000 0.98479972779514446
2000 0.99715190796798625
```
Working on it now

Reply:
what is that number? the difference? Then there is a bug

Reply:
Yes the difference. I double-checked the code, its same as scikits implementation. I'm trying to replicate this in scikit and checking how it performs.

Reply:
My guess would be that the kernel parametrization is different. Check GaussianKernel, which is something like exp(-   (x-y)^2 / with) ... while the parametrization here in the RFF is exp(-with*(x-y)^2 ...
So you need to give different bandwidth values in order to get the same result (unless the bandwidth is one). Also note that shogun accepts/stores log_width rather than with

Reply:
Lookie what I found down the rabbit hole
```
	// important trick for RFgauss to work: kernel width is set such that average inner kernel distance is close one
	// the rfgauss approximation breaks down if average inner kernel distances (~~ kernel width to small compared to variance of data) are too large
	// try rbf_width=0.1 to see how it fails! - you will see the problem in the large number of negative kernel entries (numnegratio) for the rfgauss linear kernel
```
This was under our preprocessor_randomfoureir.cpp (undocumented)

Reply:
I don't think it is causing the issue here though, your kernel width is not small (in fact you did not set it). Make sure that the kernel width is equivalent, or even yet better, convert the RFF kernel width to the same one shogun uses

Reply:
....and boom! The code is working !. Made a few interesting observations, let me tidy it up and then we can push it :D.